### PR TITLE
NXNMhPIR/2486-how-to-provide-mock-data-to-the-provider

### PR DESCRIPTION
**Pull request by mbasadi**
https://www.loom.com/share/b8ed5ab3423f42a8930537700fa41678?sid=82c952a2-f56c-4ee7-9a39-75af6c578b4a

### CODE REVIEW

**Review by sahandseifi — CHANGES_REQUESTED** [review]
There should be a mock mode where websocket, API calls, etc are not made.

Please review everywhere we mock the sdk in our front-end and ensure this can accomplish the same.

**Comment by mbasadi** [comment]
> There should be a mock mode where websocket, API calls, etc are not made.
> 
> Please review everywhere we mock the sdk in our front-end and ensure this can accomplish the same.

No network call during the mocked mode

File: lib/components/Provider/index.tsx [hunk 1]
```diff
@@ -186,15 +191,17 @@ export const NotificationAPIProvider: React.FunctionComponent<
       setLoadingNotifications(true);
 
       try {
-        await fetchNotifications(
-          initial ? new Date().toISOString() : oldestLoadedRef.current,
-          initial ? config.initialLoadMaxCount : 1000
-        );
+        if (config.connectionMode === 'live') {
```
Comment (mbasadi) [inline lib/components/Provider/index.tsx:194]: similar to https://github.com/notificationapi-com/notificationapi-js-client-sdk/blob/bb982a9aa2234453c4d379ee312d242abbab231d/src/index.ts#L247

File: lib/components/Provider/index.tsx [hunk 2]
```diff
@@ -72,6 +73,7 @@ type Props = (
   initialLoadMaxAge?: Date;
   playSoundOnNewNotification?: boolean;
   newNotificationSoundPath?: string;
+  connectionMode?: 'live' | 'mocked';
```
Comment (sahandseifi) [inline lib/components/Provider/index.tsx:76]: Please pick a better term. The mode is not about the connection (e.g. HTTP vs HTTPS), it's about the whole component and how it operates.

File: lib/components/Provider/index.tsx [hunk 3]
```diff
@@ -143,10 +146,11 @@ export const NotificationAPIProvider: React.FunctionComponent<
     });
 
     //  identify user
-    client.identify({
-      email: config.user.email,
-      number: config.user.number
-    });
+    if (config.connectionMode === 'live')
```
Comment (sahandseifi) [inline lib/components/Provider/index.tsx:149]: Please research other tools to see if it's possible to do this in a nicer way, so we don't have to use if/else for every network call.

**Review by sahandseifi — CHANGES_REQUESTED** [review]
This PR does not address the requirements of the relevant tasks. Please review the requirements and re-implement.

**Pull request closed without merging**